packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -141,49 +161,81 @@ async def unarchive_dir(
                 for zip_entry in zip_file_handler.infolist()
             ]
 
-            extracted_paths: Tuple[Path] = await asyncio.gather(*tasks)
+            extracted_paths: list[Path] = await tqdm_asyncio.gather(
+                *tasks,
+                desc=f"decompressing {archive_to_extract} -> {destination_folder} [{len(tasks)} file{'s' if len(tasks) > 1 else ''}"
+                f"/{_human_readable_size(archive_to_extract.stat().st_size)}]\n",
+                total=len(tasks),
+                unit="file",
+            )
 
             # NOTE: extracted_paths includes all tree leafs, which might include files and empty folders
-            return set(
+            return {
                 p
                 for p in extracted_paths
                 if p.is_file() or (p.is_dir() and not any(p.glob("*")))
-            )
+            }
 
 
 def _add_to_archive(
     dir_to_compress: Path,
     destination: Path,
     compress: bool,
     store_relative_path: bool,
-    exclude_patterns: List[str] = None,
+    exclude_patterns: Optional[list[str]] = None,
 ) -> Optional[Exception]:
     try:
         compression = zipfile.ZIP_DEFLATED if compress else zipfile.ZIP_STORED
         with zipfile.ZipFile(
             destination, "w", compression=compression
         ) as zip_file_handler:
-            files_to_compress_generator = _iter_files_to_compress(
-                dir_to_compress, exclude_patterns
+            folder_size_bytes = sum(
+                file.stat().st_size
+                for file in _iter_files_to_compress(dir_to_compress, exclude_patterns)
             )
-            for file_to_add in files_to_compress_generator:
-                try:
-                    file_name_in_archive = (
-                        _strip_directory_from_path(file_to_add, dir_to_compress)
-                        if store_relative_path
-                        else file_to_add
-                    )
-
-                    # because surrogates are not allowed in zip files,
-                    # replacing them will ensure errors will not happen.
-                    escaped_file_name_in_archive = _strip_undecodable_in_path(
-                        file_name_in_archive
-                    )
-
-                    zip_file_handler.write(file_to_add, escaped_file_name_in_archive)
-                except ValueError as value_error:
-                    log.exception("Could write files to archive, please check logs")
-                    raise value_error
+            desc = f"compressing {dir_to_compress} -> {destination}"
+            with tqdm(
+                desc=f"{desc}\n", total=folder_size_bytes, **_TQDM_FILE_OPTIONS
+            ) as pbar:
+                for file_to_add in _iter_files_to_compress(
+                    dir_to_compress, exclude_patterns
+                ):
+                    pbar.set_description(f"{desc}:{file_to_add.name}")
```

Comment:
MINOR: I love the ``desc`` you use. Just a small suggestion to replace ``:`` with ``/``. AFAIU  ``f"{desc}/{file_to_add.name}"`` would render

```txt
compressing /work/my/src -> /work/my/dst
compressing /work/my/src -> /work/my/dst/one.txt
compressing /work/my/src -> /work/my/dst/two.txt
```

Reply:
oups. yep that is true...